lib/ex_machina/ecto_strategy.ex

```diff
@@ -70,40 +59,76 @@ defmodule ExMachina.EctoStrategy do
     end
   end
 
-  defp cast_all_assocs(%{__struct__: schema} = struct) do
-    assocs = get_schema_assocs(schema)
+  defp cast_all_embeds(%{__struct__: schema} = struct) do
+    embed_keys = schema_embeds(schema)
 
-    Enum.reduce(assocs, struct, fn(assoc, struct) ->
-      casted_value = struct |> Map.get(assoc) |> cast_assoc(assoc, struct)
+    Enum.reduce(embed_keys, struct, fn(embed_key, struct) ->
+      casted_value = Map.get(struct, embed_key) |> cast_embed(embed_key, struct)
 
-      Map.put(struct, assoc, casted_value)
+      Map.put(struct, embed_key, casted_value)
     end)
   end
 
-  defp cast_assoc(original_assoc, assoc_key, %{__struct__: schema} = struct) do
-    case original_assoc do
-      has_or_embeds_many when is_list(has_or_embeds_many) ->
-        Enum.map(has_or_embeds_many, &(cast_assoc(&1, assoc_key, struct)))
+  defp cast_embed(embeds_many, embed_key, struct) when is_list(embeds_many) do
+    Enum.map(embeds_many, &(cast_embed(&1, embed_key, struct)))
+  end
+  defp cast_embed(embed, embed_key, %{__struct__: schema}) do
+    case embed do
+      %{} ->
+        embedding_reflection = schema.__schema__(:embed, embed_key)
+        embed_type = embedding_reflection.related
+        struct(embed_type) |> Map.merge(embed) |> cast()
+
+      nil -> nil
+    end
+  end
+
+  defp cast_all_assocs(%{__struct__: schema} = struct) do
+    assoc_keys = schema_associations(schema)
+
+    Enum.reduce(assoc_keys, struct, fn(assoc_key, struct) ->
+      casted_value = Map.get(struct, assoc_key) |> cast_assoc(assoc_key, struct)
```

Comment:
I see this in a couple places, but generally I think the accepted style is to use pipe chains when you can, 
like `struct |> Map.get(assoc_key) |> cast_assoc(assoc_key, struct)`

Reply:
This is a good catch. I tend to do this, but I think the preferred style is to have the first argument and then `|>`. I've updated this and the other mentions in b2ec4ca002be6d5a91609b45a4582e4652573e8d